proposed/0023-immutable-data-type-naming/0023-immutable-data-type-naming.md

```diff
@@ -0,0 +1,113 @@
+- Feature Name: Naming of Immutable Data Types
+- Type: Enhancement
+- Related components: [Routing](https://github.com/maidsafe/routing) and [SAFE Vault](https://github.com/maidsafe/safe_vault)
+- Start Date: 26-03-2016
+- RFC PR:
+- Issue number:
+
+# Summary
+
+For security and integrity of data, the design of the SAFE network requires that every chunk of `ImmutableData` be stored at three separate locations.  These locations are derived deterministically from the name of the chunk.  The current implementation uses repeated hashing of the chunk's name to derive the secondary names, which could result in all three locations overlapping.  This RFC proposes a solution to resolve this issue, which also has secondary benefits.
```

Comment:
I am not sure this is the primary issue as we use hash security (distribution across a range) for many elements. To me the issue is we have currently a *one way* function and the design really requires a *2 way* function. The debate about distribution of xor verses hash is secondary and more abstract with side effects of both processes.